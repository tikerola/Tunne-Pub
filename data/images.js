import image1 from "../assets/images/1.jpg";
import image2 from "../assets/images/2.jpg";
import image3 from "../assets/images/3.jpg";
import image4 from "../assets/images/4.jpg";
import image5 from "../assets/images/5.jpg";
import image6 from "../assets/images/6.jpg";
import image7 from "../assets/images/7.jpg";
import image8 from "../assets/images/8.jpg";
import image9 from "../assets/images/9.jpg";
import image10 from "../assets/images/10.jpg";
import image11 from "../assets/images/11.jpg";
import image12 from "../assets/images/12.jpg";
import image13 from "../assets/images/13.jpg";
import image14 from "../assets/images/14.jpg";
import image15 from "../assets/images/15.jpg";
import image16 from "../assets/images/16.jpg";
import image17 from "../assets/images/17.jpg";
import image18 from "../assets/images/18.jpg";
import image19 from "../assets/images/19.jpg";
import image20 from "../assets/images/20.jpg";
import image21 from "../assets/images/21.jpg";
import image22 from "../assets/images/22.jpg";
import image23 from "../assets/images/23.jpg";
import image24 from "../assets/images/24.jpg";
import image25 from "../assets/images/25.jpg";
import image26 from "../assets/images/26.jpg";
import image27 from "../assets/images/27.jpg";
import image28 from "../assets/images/28.jpg";
import image29 from "../assets/images/29.jpg";
import image30 from "../assets/images/30.jpg";
import image31 from "../assets/images/31.jpg";
import image32 from "../assets/images/32.jpg";
import image33 from "../assets/images/33.jpg";
import image34 from "../assets/images/34.jpg";
import image35 from "../assets/images/35.jpg";
import image36 from "../assets/images/36.jpg";
import image37 from "../assets/images/37.jpg";
import image38 from "../assets/images/38.jpg";
import image39 from "../assets/images/39.jpg";
import image40 from "../assets/images/40.jpg";
import image41 from "../assets/images/41.jpg";
import image42 from "../assets/images/42.jpg";
import image43 from "../assets/images/43.jpg";
import image44 from "../assets/images/44.jpg";
import image45 from "../assets/images/45.jpg";
import image46 from "../assets/images/46.jpg";
import image47 from "../assets/images/47.jpg";
import image48 from "../assets/images/48.jpg";
import image49 from "../assets/images/49.jpg";

export {
  image1,
  image2,
  image3,
  image4,
  image5,
  image6,
  image7,
  image8,
  image9,
  image10,
  image11,
  image12,
  image13,
  image14,
  image15,
  image16,
  image17,
  image18,
  image19,
  image20,
  image21,
  image22,
  image23,
  image24,
  image25,
  image26,
  image27,
  image28,
  image29,
  image30,
  image31,
  image32,
  image33,
  image34,
  image35,
  image36,
  image37,
  image38,
  image39,
  image40,
  image41,
  image42,
  image43,
  image44,
  image45,
  image46,
  image47,
  image48,
  image49,
};
