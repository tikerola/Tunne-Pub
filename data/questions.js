







export const QUESTIONS = [
  'Sinulla on ollut jokin tunteitanostattava tilanne? Kerro mitä tapahtui.',
  'Sanat, jotka parhaiten kuvaavat tunteitasi',
  'Kuvat, jotka parhaiten kuvaavat tunteitasi.',
  'Missä ja miten tunne tuntuu kehossa (esim. paino rinnan päällä, pala kurkussa, kädet menee nyrkkiin)',
  'Onko tunne alunperin sinun vai nappasitko jonkun toisen tunteen?',
  'Mikä tarve tunteen takana? Mitä olisit siinä hetkessä tarvinnut? (kuulluksi tulemista, lohdutusta, jne)',
  'Mitä tunnetta olisit kaivannut?',
  'Mikä tarve sinulla on juuri nyt?',
  'Mitä lausetta pyörität päässäsi juuri nyt? Mikä uskomus sinulla on? Mitä se sinulle merkitsee?',
  'Mitä sinun nyt pitäisi tehdä jotta voisit hyvin?'
]