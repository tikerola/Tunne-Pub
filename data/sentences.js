export const SENTENCES_DATA = [
  { id: '1', word: 'mitä muut ajattelevat?' },
  { id: '2', word: 'Olen niin huono!' },
  { id: '3', word: 'Olen väärässä.' },
  { id: '4', word: 'Joku muu tietää paremmin.' },
  { id: '5', word: 'Olen aina ennenkin antanut periksi.' },
  { id: '6', word: 'Ei tästä mitään tule, minulla ei ole rohkeutta.' },
  { id: '7', word: 'En minä osaa...' },
  { id: '8', word: 'Toinen suuttuu kuitenkin.' },
  { id: '9', word: 'Pelkään tuottaa pettymystä toiselle.' },
  { id: '10', word: 'Parempi vaan nyt olla hiljaa. Olen eri mieltä kuin muut ja pelkään tulla mielipiteeni kanssa näkyväksi.' },
  { id: '11', word: 'Tunteeni, tarpeeni, ajatukseni on huono/hölmö tai tyhmä.' }
]