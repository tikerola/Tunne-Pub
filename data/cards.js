import Card from "../models/card";
import * as images from "./images";

const CARD_DATA = [
  new Card("1", "happy", images.image1),
  new Card("2", "sad", images.image2),
  new Card("3", "worried", images.image3),
  new Card("4", "scared", images.image4),
  new Card("5", "happy", images.image5),
  new Card("6", "sad", images.image6),
  new Card("7", "worried", images.image7),
  new Card("8", "scared", images.image8),

  new Card("9", "happy", images.image9),
  new Card("10", "sad", images.image10),
  new Card("11", "worried", images.image11),
  new Card("12", "scared", images.image12),
  new Card("13", "happy", images.image13),
  new Card("14", "sad", images.image14),
  new Card("15", "worried", images.image15),
  new Card("16", "scared", images.image16),

  new Card("17", "happy", images.image17),
  new Card("18", "sad", images.image18),
  new Card("19", "worried", images.image19),
  new Card("20", "scared", images.image20),
  new Card("21", "happy", images.image21),
  new Card("22", "sad", images.image22),
  new Card("23", "worried", images.image23),
  new Card("24", "scared", images.image24),

  new Card("25", "happy", images.image25),
  new Card("26", "sad", images.image26),
  new Card("27", "worried", images.image27),
  new Card("28", "scared", images.image28),
  new Card("29", "happy", images.image29),
  new Card("30", "sad", images.image30),
  new Card("31", "worried", images.image31),
  new Card("32", "scared", images.image32),

  new Card("33", "happy", images.image33),
  new Card("34", "sad", images.image34),
  new Card("35", "worried", images.image35),
  new Card("36", "scared", images.image36),
  new Card("37", "happy", images.image37),
  new Card("38", "sad", images.image38),
  new Card("39", "worried", images.image39),
  new Card("40", "scared", images.image40),

  new Card("41", "happy", images.image41),
  new Card("42", "sad", images.image42),
  new Card("43", "worried", images.image43),
  new Card("44", "scared", images.image44),
  new Card("45", "happy", images.image45),
  new Card("46", "sad", images.image46),
  new Card("47", "worried", images.image47),
  new Card("48", "scared", images.image48),
  new Card("49", "scared", images.image49),
];

export default CARD_DATA;
