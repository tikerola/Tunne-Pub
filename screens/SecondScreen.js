//import { ProgressStep, ProgressSteps } from 'react-native-progress-steps'
import { ProgressStep } from "@tkkortit/own-stepper";
import React, { useEffect, useRef, useState } from "react";
import {
  ActivityIndicator,
  Dimensions,
  ScrollView,
  StyleSheet,
  Text,
  View,
} from "react-native";
import { useDispatch, useSelector } from "react-redux";
import CustomHeaderTitle from "../components/CustomHeaderTitle";
import CustomProgressStep from "../components/CustomProgressStep";
import CustomProgressSteps from "../components/CustomProgressSteps";
import ProgressModeStepContent from "../components/ProgressModeStepContent";
import ProgressStepContent from "../components/ProgressStepContent";
import { colors } from "../constants/colors";
import { dimensions } from "../constants/dimensions";
import { QUESTIONS } from "../data/questions";
import { addAnswerToAQuestion, setScreenNumber } from "../store/actions/cards";

const defaultScrollViewProps = {
  keyboardShouldPersistTaps: "handled",
  contentContainerStyle: {
    flex: 1,
    justifyContent: "center",
  },
};

const START_STEP = 4;

export default function SecondScreen({ route, navigation }) {
  const [answer, setAnswer] = useState("");
  const [activeIndex, setActiveIndex] = useState(START_STEP);
  const [error, setError] = useState("");
  const [listMode, setListMode] = useState(true);

  const dispatch = useDispatch();
  const { height, width } = Dimensions.get("window");

  const summaryId = useSelector((state) => state.cards.activeSummaryItemId);

  const summaryItem = useSelector(
    (state) => state.cards.summaryOfChosenItems[state.cards.activeSummaryItemId]
  );
  const screenNumber = useSelector((state) => state.cards.screenNumber);

  const progressRef = useRef();

  const { name } = route.params;

  useEffect(() => {
    navigation.setOptions({
      headerTitle: (props) => (
        <CustomHeaderTitle {...props} letterNum={4} name={name} />
      ),
    });
  }, [name]);

  useEffect(() => {
    setAnswer("");
    setListMode(true);
    if (summaryItem && summaryItem.questions[QUESTIONS[activeIndex]])
      populateFields(activeIndex);
  }, [activeIndex, screenNumber]);

  const populateFields = (activeIndex) => {
    if (activeIndex >= 4 && activeIndex <= 6)
      setAnswer(summaryItem.questions[QUESTIONS[activeIndex]]);
  };

  const handleSubmit = (answerToAQuestion, wordType) => {
    if (!answerToAQuestion && !summaryItem.questions[QUESTIONS[activeIndex]]) {
      setError("Et voi tallentaa tyhjää sisältöä!");
      setTimeout(() => setError(""), 2500);
      return;
    }

    if (activeIndex >= 4 && activeIndex <= 6)
      dispatch(
        addAnswerToAQuestion(
          summaryId,
          QUESTIONS[activeIndex],
          answerToAQuestion,
          listMode,
          wordType
        )
      );

    if (activeIndex < 6) {
      progressRef.current.setActiveStep(activeIndex - (START_STEP - 1) + 1);
      setActiveIndex((prev) => prev + 1);
    } else {
      dispatch(setScreenNumber(3));
      navigation.navigate("ThirdScreen", { name });
    }
  };

  const goToPreviousPage = () => {
    dispatch(setScreenNumber(1));
    navigation.navigate("MainScreen");
  };

  const goToNextPage = () => {
    dispatch(setScreenNumber(3));
    navigation.navigate("ThirdScreen", { name });
  };

  const getRef = (ref) => {
    progressRef.current = ref;
  };

  if (!summaryItem)
    return (
      <View
        style={{
          flex: 1,
          alignItems: "center",
          backgroundColor: colors.bodyBg,
          justifyContent: "center",
        }}
      >
        <ActivityIndicator testID="activity-indicator" size="large" />
      </View>
    );

  return (
    <ScrollView
      keyboardShouldPersistTaps={"handled"}
      contentContainerStyle={{ flexGrow: 1 }}
    >
      <View style={styles.container}>
        <View
          style={{
            width: width <= 500 ? "100%" : "60%",
            height: height <= 750 ? "100%" : "75%",
          }}
        >
          <CustomProgressSteps
            getRef={getRef}
            startStep={START_STEP}
            activeStep={1}
          >
            <ProgressStep label="kehon tuntemukset">
              <Text>Dummy Step</Text>
            </ProgressStep>

            <CustomProgressStep
              label="tunteen alkuperä"
              onPrevious={goToPreviousPage}
              onNext={() => setActiveIndex((prevIndex) => prevIndex + 1)}
            >
              <ProgressStepContent
                description={QUESTIONS[activeIndex]}
                answer={answer}
                setAnswer={setAnswer}
                error={error}
                handleSubmit={handleSubmit}
                pad={height > dimensions.phoneMaxHeight}
              />
            </CustomProgressStep>

            <CustomProgressStep
              onNext={() => setActiveIndex((prevIndex) => prevIndex + 1)}
              onPrevious={() => setActiveIndex((prevIndex) => prevIndex - 1)}
              label="tarve tunteen takana"
            >
              <ProgressModeStepContent
                navigation={navigation}
                navigateTo="WordScreen"
                wordType="needsBehindAFeeling"
                navigateBackTo="SecondScreen"
                question={5}
                listMode={listMode}
                setListMode={setListMode}
                handleSubmit={handleSubmit}
                description={QUESTIONS[activeIndex]}
                answer={answer}
                setAnswer={setAnswer}
                error={error}
                pad={height > dimensions.phoneMaxHeight}
              />
            </CustomProgressStep>

            <CustomProgressStep
              label="kaipaamasi tunne"
              onNext={() => setActiveIndex((prevIndex) => prevIndex + 1)}
              onPrevious={() => setActiveIndex((prevIndex) => prevIndex - 1)}
              onSubmit={goToNextPage}
            >
              <ProgressModeStepContent
                navigation={navigation}
                navigateTo="WordScreen"
                wordType="feelingsYouMissWords"
                navigateBackTo="SecondScreen"
                question={6}
                listMode={listMode}
                setListMode={setListMode}
                handleSubmit={handleSubmit}
                description={QUESTIONS[activeIndex]}
                answer={answer}
                setAnswer={setAnswer}
                error={error}
                pad={height > dimensions.phoneMaxHeight}
              />
            </CustomProgressStep>
          </CustomProgressSteps>
        </View>
      </View>
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: colors.bodyBg,
  },

  buttonContainer: {
    marginTop: 5,
    flexDirection: "row",
    justifyContent: "space-around",
  },

  buttonStyle: {
    color: "#bbb",
    fontSize: 18,
    fontFamily: "Roboto-Light",
  },

  listButtonText: {
    fontSize: 18,
    // color: 'white',
    //borderRadius: 5,
    // padding: 4,
    textDecorationLine: "underline",
    //backgroundColor: 'black'
  },

  instructionText: {
    fontSize: 18,
    padding: 15,
  },

  bolded: {
    fontWeight: "bold",
    fontSize: 18,

    marginBottom: 5,
  },

  image: {
    width: 96,
    height: 140,
    marginRight: 5,
  },
  error: {
    textAlign: "center",
    fontSize: 16,
    fontFamily: "Roboto-Bold",
    color: "red",
  },
  pad: {
    backgroundColor: colors.stepperContentBg,
    padding: 50,
    borderRadius: 10,
  },
});
