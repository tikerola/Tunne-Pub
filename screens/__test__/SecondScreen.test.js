import "@testing-library/jest-native/extend-expect";
import { fireEvent } from "@testing-library/react-native";
import React from "react";
import configureStore from "redux-mock-store";
import { QUESTIONS } from "../../data/questions";
import {
  act,
  store,
  render,
  waitFor,
  waitForElementToBeRemoved,
} from "../../test-utils";
import SecondScreen from "../SecondScreen";

describe("<SecondScreen />", () => {
  const INITIAL_STATE = {
    cards: {
      activeSummaryItemId: "",
      summaryOfChosenItems: {
        123: {
          id: "123",
          name: "123",
          cards: [],
          feelingsWords: ["pahoillaan", "yskä"],
          feelingsYouMissWords: [],
          needs: ["hei"],
          needsBehindAFeeling: [],
          sentences: [],
          questions: { [QUESTIONS[0]]: "niinpä niin" },
          itemSaved: true,
        },
      },
      screenNumber: 1,
      numOfItems: 0,
      initialLoadFinished: false,
    },
  };

  const INITIAL_STATE2 = {
    cards: {
      activeSummaryItemId: "123",
      summaryOfChosenItems: {
        123: {
          id: "123",
          name: "",
          cards: [],
          feelingsWords: ["pahoillaan", "yskä"],
          feelingsYouMissWords: [],
          needs: ["hei"],
          needsBehindAFeeling: [],
          sentences: [],
          questions: {
            [QUESTIONS[4]]: "niinpä niin",
            [QUESTIONS[5]]: "pahoillaan, yskä",
            [QUESTIONS[6]]: "diipadaapa",
          },
          itemSaved: true,
        },
      },
      screenNumber: 1,
      numOfItems: 0,
      initialLoadFinished: false,
    },
  };

  const INITIAL_STATE3 = {
    cards: {
      activeSummaryItemId: "123",
      summaryOfChosenItems: {},
      screenNumber: 41,
      numOfItems: 0,
      initialLoadFinished: false,
    },
  };

  const navigation = {
    navigate: jest.fn(),
    setOptions: jest.fn(),
  };

  route = {
    params: {
      name: "Hei",
    },
  };

  const mockStore = configureStore([]);

  it("should match snapshot", () => {
    const result = render(
      <SecondScreen navigation={navigation} route={route} />
    ).toJSON();
    expect(result).toMatchSnapshot();
  });

  it("should set navigation options when name provided by route params", async () => {
    const component = render(
      <SecondScreen navigation={navigation} route={route} />,
      undefined,
      mockStore(INITIAL_STATE2)
    );

    expect(navigation.setOptions).toHaveBeenCalled();
  });

  it("should prepopulate 5. field when updating a record", async () => {
    const store = mockStore(INITIAL_STATE2);

    const component = render(
      <SecondScreen navigation={navigation} route={route} />,
      undefined,
      store
    );
    expect(component.queryByText(QUESTIONS[4])).toBeTruthy();
    expect(
      component.queryByPlaceholderText("Kirjoita omin sanoin...")
    ).toBeTruthy();

    let input = component.queryByTestId("custom-input");
    expect(input.props.value).toEqual("niinpä niin");
  });

  it("should prepopulate 6. field when updating a record", async () => {
    const store = mockStore(INITIAL_STATE2);

    const component = render(
      <SecondScreen navigation={navigation} route={route} />,
      undefined,
      store
    );

    fireEvent.press(component.getByText("Seuraava"));
    expect(
      await waitFor(() => component.queryByText(QUESTIONS[5]))
    ).toBeTruthy();

    expect(
      component.queryByPlaceholderText(
        "Paina minua lisätäksesi sanoja sanalistasta..."
      )
    ).toBeTruthy();

    let input = component.queryByTestId("custom-input");
    expect(input.props.value).toEqual("pahoillaan, yskä");

    fireEvent.press(input);
    const actions = await waitFor(() => store.getActions());
    expect(actions[0].type).toEqual("SET SCREEN NUMBER");

    expect(navigation.navigate).toHaveBeenCalledWith("WordScreen", {
      navigateBackTo: "SecondScreen",
      question: 5,
      wordType: "needsBehindAFeeling",
    });
  });

  it("should prepopulate 7. field when updating a record", async () => {
    const store = mockStore(INITIAL_STATE2);

    const component = render(
      <SecondScreen navigation={navigation} route={route} />,
      undefined,
      store
    );

    fireEvent.press(component.getByText("Seuraava"));
    fireEvent.press(component.getByText("Seuraava"));

    expect(
      await waitFor(() => component.queryByText(QUESTIONS[6]))
    ).toBeTruthy();

    expect(
      component.queryByPlaceholderText(
        "Paina minua lisätäksesi sanoja sanalistasta..."
      )
    ).toBeTruthy();

    let input = component.queryByTestId("custom-input");
    expect(input.props.value).toEqual("diipadaapa");

    fireEvent.press(input);

    expect(navigation.navigate).toHaveBeenCalledWith("WordScreen", {
      navigateBackTo: "SecondScreen",
      question: 6,
      wordType: "feelingsYouMissWords",
    });
  });

  it("should go to next index after having saved an answer", async () => {
    const component = render(
      <SecondScreen navigation={navigation} route={route} />
    );

    expect(component.queryByText(QUESTIONS[4])).toBeTruthy();
    const input = component.getByTestId("custom-input");
    fireEvent.changeText(input, "");
    fireEvent.press(component.getByText("Tallenna"));
    expect(
      component.queryByText("Et voi tallentaa tyhjää sisältöä!")
    ).toBeTruthy();

    await waitForElementToBeRemoved(() =>
      component.getByText("Et voi tallentaa tyhjää sisältöä!")
    );

    fireEvent.changeText(input, "Some Answer");
    fireEvent.press(component.getByText("Tallenna"));

    expect(component.queryByText(QUESTIONS[4])).toBeFalsy();
    expect(component.queryByText(QUESTIONS[5])).toBeTruthy();

    const input2 = component.getByTestId("custom-input");
    expect(input2.props.editable).toEqual(false);

    const toggle = component.queryByTestId("switch");
    await waitFor(() => fireEvent.press(toggle));

    expect(input2.props.editable).toEqual(true);

    fireEvent.changeText(input2, "Some Answer");
    fireEvent.press(component.getByText("Tallenna"));

    expect(component.queryByText(QUESTIONS[5])).toBeFalsy();
    expect(component.queryByText(QUESTIONS[6])).toBeTruthy();

    const input3 = component.getByTestId("custom-input");
    fireEvent.changeText(input3, "Some Answer");
    fireEvent.press(component.getByText("Tallenna"));

    const actions = store.getActions();
    expect(actions).toContainEqual({
      type: "SET SCREEN NUMBER",
      screenNumber: 3,
    });
    expect(navigation.navigate).toHaveBeenCalledWith("ThirdScreen", {
      name: "Hei",
    });
  });

  it("should test next and previous buttons", async () => {
    const component = render(
      <SecondScreen navigation={navigation} route={route} />
    );

    expect(component.queryByText(QUESTIONS[4])).toBeTruthy();
    fireEvent.press(component.getByText("Seuraava"));

    expect(
      await waitFor(() => component.queryByText(QUESTIONS[5]))
    ).toBeTruthy();
    fireEvent.press(component.getByText("Seuraava"));

    expect(
      await waitFor(() => component.queryByText(QUESTIONS[6]))
    ).toBeTruthy();
    fireEvent.press(component.getByText("Seuraava"));

    expect(navigation.navigate).toHaveBeenCalledWith("ThirdScreen", {
      name: "Hei",
    });

    fireEvent.press(component.getByText("Edellinen"));
    expect(
      await waitFor(() => component.queryByText(QUESTIONS[5]))
    ).toBeTruthy();

    fireEvent.press(component.getByText("Edellinen"));
    expect(
      await waitFor(() => component.queryByText(QUESTIONS[4]))
    ).toBeTruthy();

    fireEvent.press(component.getByText("Edellinen"));
    expect(navigation.navigate).toHaveBeenCalledWith("MainScreen");
  });

  it("should show activityIndicator if summaryItem has not loaded", async () => {
    const component = render(
      <SecondScreen navigation={navigation} route={route} />,
      undefined,
      mockStore(INITIAL_STATE3)
    );

    expect(component.queryByTestId("activity-indicator")).toBeTruthy();
  });
});
