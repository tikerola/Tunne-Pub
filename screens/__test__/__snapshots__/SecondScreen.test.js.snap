// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SecondScreen /> should match snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
  keyboardShouldPersistTaps="handled"
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5f0c38",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "height": "75%",
            "width": "60%",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "flexDirection": "row",
                "justifyContent": "space-evenly",
                "marginBottom": 200,
                "position": "relative",
                "top": 30,
              }
            }
          >
            <View>
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "column",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "red",
                        "borderRadius": 18,
                        "height": 36,
                        "width": 36,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "alignSelf": "center",
                          "top": 9,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "white",
                          }
                        }
                      >
                        ✓
                      </Text>
                    </Text>
                  </View>
                  <Text
                    style={
                      Object {
                        "color": "lightgray",
                        "flexWrap": "wrap",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "marginTop": 4,
                        "paddingTop": 4,
                        "textAlign": "center",
                        "width": 100,
                      }
                    }
                  >
                    kehon tuntemukset
                  </Text>
                  <View
                    style={
                      Object {
                        "borderTopColor": "red",
                        "borderTopStyle": "solid",
                        "borderTopWidth": 3,
                        "left": 44,
                        "marginLeft": 22,
                        "position": "absolute",
                        "right": 0,
                        "top": 18,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View>
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "column",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "borderColor": "red",
                        "borderRadius": 20,
                        "borderWidth": 5,
                        "bottom": 2,
                        "height": 40,
                        "width": 40,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "alignSelf": "center",
                          "top": 6.666666666666667,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#bbb",
                          }
                        }
                      >
                        5
                      </Text>
                    </Text>
                  </View>
                  <Text
                    style={
                      Object {
                        "color": "#4BB543",
                        "flexWrap": "wrap",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "paddingTop": 4,
                        "textAlign": "center",
                        "width": 100,
                      }
                    }
                  >
                    tunteen alkuperä
                  </Text>
                  <View
                    style={
                      Object {
                        "borderTopColor": "red",
                        "borderTopStyle": "solid",
                        "borderTopWidth": 3,
                        "left": 0,
                        "marginRight": 22,
                        "position": "absolute",
                        "right": 48,
                        "top": 18.018018018018015,
                      }
                    }
                  />
                  <View
                    style={
                      Object {
                        "borderTopColor": "black",
                        "borderTopStyle": "solid",
                        "borderTopWidth": 3,
                        "left": 48,
                        "marginLeft": 22,
                        "position": "absolute",
                        "right": 0,
                        "top": 18.018018018018015,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View>
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "column",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "black",
                        "borderRadius": 18,
                        "height": 36,
                        "width": 36,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "alignSelf": "center",
                          "top": 9,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "white",
                          }
                        }
                      >
                        6
                      </Text>
                    </Text>
                  </View>
                  <Text
                    style={
                      Object {
                        "color": "lightgray",
                        "flexWrap": "wrap",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "marginTop": 4,
                        "paddingTop": 4,
                        "textAlign": "center",
                        "width": 100,
                      }
                    }
                  >
                    tarve tunteen takana
                  </Text>
                  <View
                    style={
                      Object {
                        "borderTopColor": "black",
                        "borderTopStyle": "solid",
                        "borderTopWidth": 3,
                        "left": 0,
                        "marginRight": 22,
                        "position": "absolute",
                        "right": 44,
                        "top": 18,
                      }
                    }
                  />
                  <View
                    style={
                      Object {
                        "borderTopColor": "black",
                        "borderTopStyle": "solid",
                        "borderTopWidth": 3,
                        "left": 44,
                        "marginLeft": 22,
                        "position": "absolute",
                        "right": 0,
                        "top": 18,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View>
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "column",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "black",
                        "borderRadius": 18,
                        "height": 36,
                        "width": 36,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "alignSelf": "center",
                          "top": 9,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "white",
                          }
                        }
                      >
                        7
                      </Text>
                    </Text>
                  </View>
                  <Text
                    style={
                      Object {
                        "color": "lightgray",
                        "flexWrap": "wrap",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "marginTop": 4,
                        "paddingTop": 4,
                        "textAlign": "center",
                        "width": 100,
                      }
                    }
                  >
                    kaipaamasi tunne
                  </Text>
                  <View
                    style={
                      Object {
                        "borderTopColor": "black",
                        "borderTopStyle": "solid",
                        "borderTopWidth": 3,
                        "left": 0,
                        "marginRight": 22,
                        "position": "absolute",
                        "right": 44,
                        "top": 18,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <RCTScrollView
                keyboardShouldPersistTaps="handled"
              >
                <View>
                  <View
                    style={
                      Object {
                        "backgroundColor": "#440627",
                        "borderRadius": 10,
                        "padding": 50,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "display": "flex",
                          "flexDirection": "column",
                          "flexGrow": 1,
                          "marginTop": 0,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#bbb",
                            "fontFamily": "Roboto-Light",
                            "fontSize": 18,
                            "width": "95%",
                          }
                        }
                      >
                        Onko tunne alunperin sinun vai nappasitko jonkun toisen tunteen?
                      </Text>
                      <TextInput
                        allowFontScaling={true}
                        keyboardAppearance="dark"
                        multiline={true}
                        numberOfLines={5}
                        onChangeText={[Function]}
                        placeholder="Kirjoita omin sanoin..."
                        placeholderTextColor="#555"
                        rejectResponderTermination={true}
                        style={
                          Object {
                            "backgroundColor": "#333",
                            "borderColor": "#bbb",
                            "borderRadius": 5,
                            "borderWidth": 1,
                            "color": "#bbb",
                            "padding": 5,
                            "width": "95%",
                          }
                        }
                        testID="custom-input"
                        textAlignVertical="top"
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </View>
                    <Text
                      style={
                        Object {
                          "color": "red",
                          "fontFamily": "Roboto-Bold",
                          "fontSize": 16,
                          "textAlign": "center",
                        }
                      }
                    >
                      
                    </Text>
                    <View
                      accessible={true}
                      focusable={true}
                      isTVSelectable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignSelf": "center",
                          "backgroundColor": "#3a89ed",
                          "borderRadius": 5,
                          "elevation": 5,
                          "margin": 0,
                          "opacity": 1,
                          "padding": 5,
                          "shadowColor": "#000",
                          "shadowOffset": Object {
                            "height": 4,
                            "width": 0,
                          },
                          "shadowOpacity": 0.44,
                          "shadowRadius": 5,
                          "width": 100,
                        }
                      }
                      testID="button"
                    >
                      <Text
                        style={
                          Object {
                            "alignSelf": "center",
                            "color": "white",
                            "fontFamily": "Roboto-Light",
                            "fontSize": 16,
                          }
                        }
                        testID="title"
                      >
                        Tallenna
                      </Text>
                    </View>
                  </View>
                </View>
              </RCTScrollView>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "marginTop": 90,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "bottom": 40,
                      "left": 60,
                      "position": "absolute",
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                        "padding": 8,
                        "textAlign": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#bbb",
                            "fontFamily": "Roboto-Light",
                            "fontSize": 18,
                          },
                        ]
                      }
                    >
                      Edellinen
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "bottom": 40,
                      "position": "absolute",
                      "right": 60,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                        "padding": 8,
                        "textAlign": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#bbb",
                            "fontFamily": "Roboto-Light",
                            "fontSize": 18,
                          },
                        ]
                      }
                    >
                      Seuraava
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
